### PR TITLE
[Bugfix][AMD] Fix memory access fault and update document

### PR DESCRIPTION
**Pull request by hnts03-moreh**
Co-authored-by: HakJu Kim <hakju.kim@moreh.io>

## Motivation
This PR is for bugfix and doc update. 
* Fixing memory access fault issue reported at #10195 .
* Following current `docs/platforms/amd_gpu.md` document causes installing unintended dependencies such as `nvidia-cuda-runtime-cu12-12.8.90` package due to out-dated guide. (Related PR: #10465)

## Modifications
There are two modifications.
1. Filter and change the value of topk_ids in `forward()` function of `FusedMoE` class, `-1` to self.local_num_experts`.
2. Update the install guide for AMD devices to represent current file structure.

## Accuracy Tests

Modification of topk_ids could make correctness issue so we tried accuracy test.
```
# python3 ./few_shot_gsm8k.py --num-questions 200
Downloading from https://raw.githubusercontent.com/openai/grade-school-math/master/grade_school_math/data/test.jsonl to /tmp/test.jsonl
/tmp/test.jsonl: 732kB [00:00, 9.31MB/s]
100%|████████████████████████████████████████████████████████████████████████████████████████| 200/200 [00:39<00:00,  5.11it/s]
Accuracy: 0.970
Invalid: 0.000
Latency: 53.490 s
Output throughput: 342.343 token/s
```

## Benchmarking and Profiling

This PR is not about speed-up or profiling features.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Review by hubertlu-tw — APPROVED** [review]
LGTM

File: python/sglang/srt/layers/moe/fused_moe_triton/layer.py [hunk 1]
```diff
@@ -823,6 +823,17 @@ def forward(self, hidden_states: torch.Tensor, topk_output: TopKOutput):
...
+            # NOTE: Aiter's fused_moe does not support '-1' value which is broadly used to represent the invalid expert id.
+            # If '-1' is given to gpu kernel, it may cause memory access fault.
+            if TopKOutputChecker.format_is_standard(topk_output):
+                new_topk_ids = torch.where(
+                    topk_output.topk_ids == -1,
+                    self.num_local_experts,
+                    topk_output.topk_ids,
+                )
+                topk_output = topk_output._replace(topk_ids=new_topk_ids)
+
```
Comment (HaiShaw) [inline python/sglang/srt/layers/moe/fused_moe_triton/layer.py]: Better move `if _use_aiter:` to `if TopKOutputChecker.format_is_standard(topk_output):` clause above (#L819)
Please confirm that `self.num_local_experts` will be the same size as `topk_output.topk_ids`

File: docs/platforms/amd_gpu.md [hunk 2]
```diff
@@ -53,6 +53,8 @@ cd sgl-kernel
 python setup_rocm.py install
 
 # Install sglang python package
+cd ../python
+rm -rf pyproject.toml && mv pyproject_other.toml pyproject.toml
```
[the remark this answers is not in the document]
Comment (HaiShaw) [reply docs/platforms/amd_gpu.md]: Please pick up this suggestion

[the remark this answers is not in the document]
Comment (hnts03-moreh) [reply docs/platforms/amd_gpu.md]: Already applied at latest commit.

File: python/sglang/srt/layers/moe/fused_moe_triton/layer.py (hunk 1 above)
Comment (hnts03-moreh) [reply python/sglang/srt/layers/moe/fused_moe_triton/layer.py]: Thanks for your review.

> Better move `if _use_aiter:` to `if TopKOutputChecker.format_is_standard(topk_output):` clause above (#L819)

I'll move the `if _use_aiter:` block to #L819.

> Please confirm that `self.num_local_experts` will be the same size as `topk_output.topk_ids`

There is no relevance between `self.num_local_experts` and size of `topk_output.topk_ids` I think. This code changes `-1` value inside of `topk_output.topk_ids` into `self.num_local_experts` to avoid memory access fault on aiter's fused_moe kernel. 
Could you explain more about this review?

**Comment by HaiShaw** [comment]
Is this `I'll move the if _use_aiter: block to #L819.` done?

**Comment by hnts03-moreh** [comment]
@HaiShaw 
I've pushed the fix for the code block move in the latest commit ([8137064](https://github.com/sgl-project/sglang/pull/11154/commits/81370643bb127313cdc3d0287183bdd7325e04c6)).

File: python/sglang/srt/layers/moe/fused_moe_triton/layer.py [hunk 3]
```diff
@@ -820,6 +820,18 @@ def forward(self, hidden_states: torch.Tensor, topk_output: TopKOutput):
                 topk_output = topk_output._replace(
                     topk_ids=self.expert_map_gpu[topk_output.topk_ids]
                 )
+
+                if _use_aiter:
+                    # NOTE: Aiter's fused_moe does not support '-1' value which is broadly used to represent the invalid expert id.
+                    # If '-1' is given to gpu kernel, it may cause memory access fault.
+                    if TopKOutputChecker.format_is_standard(topk_output):
```
Comment (HaiShaw) [inline python/sglang/srt/layers/moe/fused_moe_triton/layer.py]: why do this `if` check again ?

**Review by HaiShaw — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/moe/fused_moe_triton/layer.py (hunk 3 above)
Comment (hnts03-moreh) [reply python/sglang/srt/layers/moe/fused_moe_triton/layer.py]: My fault. I'll delete it. 
Is there any code which need to be changed else?

**Review by chhzh123 — APPROVED** [review]
LGTM. Can you add unit tests for this if possible?

**Comment by hnts03-moreh** [comment]
@chhzh123 
Thank you for the review.
This PR currently has a major conflict with [Cheng Wan's MoE refactoring](https://github.com/sgl-project/sglang/pull/11847). Resolving it is taking a bit longer than we expected. I'll let you know as soon as it's done.
